### PR TITLE
Graduate API Server tracing to beta

### PR DESCRIPTION
**Pull request by dashpole**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Graduate the APIServerTracing feature to beta.  This updates the feature-gate to beta, enables it by default, and adds TracingConfiguration to the v1beta1 config group. 

This is the second attempt at graduation see https://github.com/kubernetes/kubernetes/pull/113693 for the original PR.  The original PR introduced a bug, https://github.com/kubernetes/kubernetes/issues/113791, which broke port-forwarding.  Before it can be merged, this PR must:
* [x] be rebased on https://github.com/kubernetes/kubernetes/pull/116140
* [x] consistently pass `[sig-instrumentation] MetricsGrabber` tests to ensure the original bug is not reintroduced.

Beta requirements from KEP:

* [x] Tracing 100% of requests does not break scalability tests (this does not necessarily mean trace backends can handle all the data).
    * Testing in: https://github.com/kubernetes/kubernetes/pull/113695
    * Test results: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/113695/pull-kubernetes-e2e-gce-scale-performance-manual/1589994122503000064
* [x] OpenTelemetry reaches GA: https://github.com/kubernetes/kubernetes/pull/112545
* [x] Publish examples of how to use the OT Collector with kubernetes: https://kubernetes.io/docs/concepts/cluster-administration/system-traces/#trace-collection
* [x] Allow time for feedback
* [x] Revisit the format used to export spans.
* [x] Parity with the old text-based Traces: https://github.com/kubernetes/kubernetes/issues/113170

#### Does this PR introduce a user-facing change?

Release-notes are removed, since it is reverted
```release-note
APIServerTracing feature gate is now enabled by default. Tracing in the API Server is still disabled by default, and requires a config file to enable.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/bddca24910fb349e2eb0ac1c822c77f0f32fe9c6/keps/sig-instrumentation/647-apiserver-tracing/README.md
```

@kubernetes/sig-instrumentation-leads

### CODE REVIEW

**Comment by dashpole** [comment]
cc @kubernetes/sig-instrumentation-leads

**Comment by dashpole** [comment]
cc @liggitt, who reviewed the original pr (https://github.com/kubernetes/kubernetes/pull/113693)

**Comment by liggitt** [comment]
was https://github.com/kubernetes/kubernetes/pull/116140 the only change needed? is this a replay of the previous graduation PR otherwise?

**Comment by dashpole** [comment]
Yes.  See https://github.com/kubernetes/kubernetes/issues/113791#issuecomment-1309391196.  It was just so close to the release that I thought it better not to try any heroics.  I was hoping to just wait for a regular release from otel, but that ended up being blocked by https://github.com/kubernetes/kubernetes/issues/113366.  So the otel-go maintainers agreed to make a patch release for me with just this fix: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/3513

**Comment by liggitt** [comment]
the otel bump is merged if you want to rebase

**Comment by dashpole** [comment]
Rebased.  I confirmed that MetricsGrabber tests pass consistently (~20 times) against this PR:

```
$ kubetest --test --test_args="--ginkgo.focus=MetricsGrabber --ginkgo.v=true"
...
Ran 5 of 7120 Specs in 12.155 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 7115 Skipped
PASS

Ginkgo ran 1 suite in 12.510329816s
Test Suite Passed
2023/03/04 17:19:33 process.go:155: Step './hack/ginkgo-e2e.sh --ginkgo.focus=MetricsGrabber --ginkgo.v=true' finished in 15.309633433s
```

**Comment by dashpole** [comment]
```
=== FAIL: vendor/k8s.io/apimachinery/pkg/util/httpstream/spdy TestConnectionPings (0.03s)
    connection_test.go:240: server: failed to send any pings (check logs)
```

Seems unrelated.  Looks like https://github.com/kubernetes/kubernetes/issues/116165

/retest

**Comment by liggitt** [comment]
> ```
> === FAIL: vendor/k8s.io/apimachinery/pkg/util/httpstream/spdy TestConnectionPings (0.03s)
>     connection_test.go:240: server: failed to send any pings (check logs)
> ```
> 
> 
>     
>   
> 
> Seems unrelated.  Looks like #116165

cc @aojea - fix didn't work 😢

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ const (
 	APIServerIdentity featuregate.Feature = "APIServerIdentity"
 
 	// owner: @dashpole
-	// alpha: v1.22
+	// beta: v1.26
```
Comment (pacoxu) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go:76]:
```suggestion
	// beta: v1.27
```

LGTM  except this.

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/features/kube_features.go:76]: yeah, also keep the alpha line so we can see the feature progress in file

**Comment by liggitt** [comment]
lgtm once https://github.com/kubernetes/kubernetes/pull/116144/files#r1127412920 is tweaked

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go (hunk 1 above)
Comment (dashpole) [reply staging/src/k8s.io/apiserver/pkg/features/kube_features.go:76]: done, and done.

**Comment by pacoxu** [comment]
/retest
Test_StorageVersionUpdatedWithAllEncodingVersionsEqualOnLeaseDeletion flake?

**Pull request merged**